src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.cs

```diff
@@ -369,34 +369,21 @@ private Task AddOrUpdateProjectCoreAsync(
                     _logger.LogInformation($"Updating project '{project.Key}' TagHelpers ({projectWorkspaceState.TagHelpers.Length}) and C# Language Version ({projectWorkspaceState.CSharpLanguageVersion}).");
                 }
 
-                updater.ProjectWorkspaceStateChanged(project.Key, projectWorkspaceState);
-
                 var currentConfiguration = project.Configuration;
                 var currentRootNamespace = project.RootNamespace;
-                if (currentConfiguration.ConfigurationName == configuration?.ConfigurationName &&
-                    currentRootNamespace == rootNamespace)
-                {
-                    _logger.LogTrace($"Updating project '{project.Key}'. The project is already using configuration '{configuration.ConfigurationName}' and root namespace '{rootNamespace}'.");
-                    return;
-                }
-
                 if (configuration is null)
                 {
                     configuration = FallbackRazorConfiguration.Latest;
                     _logger.LogInformation($"Updating project '{project.Key}' to use the latest configuration ('{configuration.ConfigurationName}')'.");
                 }
-                else if (currentConfiguration.ConfigurationName != configuration.ConfigurationName)
-                {
-                    _logger.LogInformation($"Updating project '{project.Key}' to Razor configuration '{configuration.ConfigurationName}' with language version '{configuration.LanguageVersion}'.");
-                }
-
-                if (currentRootNamespace != rootNamespace)
+                else if (currentConfiguration.Equals(configuration) &&
+                    currentRootNamespace == rootNamespace)
```

Comment:
Isn't `RazorConfiguration` is a record? If so, it should be equivalent to use `==`.

```suggestion
                else if (currentConfiguration == configuration &&
                    currentRootNamespace == rootNamespace)
```

Reply:
Yes, you're right. Paranoia because `ProjectWorkspaceState` isn't a record, yet we had code that did `==` and `.Equals` and I deleted one of the checks, but it was the wrong one. Looking forward to fixing ProjectWorkspaceState in future.

Reply:
I've got a few branches with `ProjectWorkspaceState` converted to a record that I'm hoping to resurrect.